File: Runtime/TransitionGroup.vb
'*********************************************************************************************
' ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'*********************************************************************************************

Public Class TransitionGroup

    Private m_TransitionGroupId As Integer
    Private m_DisplayName As String
    Private m_TransitionTypes As New TransitionTypeCollection
    Private m_PrimaryTransitionTypes As New TransitionTypeCollection
    Private m_TransitionSpreadCells As New Dictionary(Of Integer, Cell)
    Private m_TransitionSpreadDistributionMap As New TransitionSpreadDistributionMap()
    Private m_HasSizeDistribution As Boolean
    Private m_PatchPrioritization As PatchPrioritization
    Private m_Order As Double = DEFAULT_TRANSITION_ORDER
    Private m_IsAuto As Boolean

    Public Sub New(
        ByVal transitionGroupId As Integer,
        ByVal transitionGroupName As String,
        ByVal isAuto As Boolean)

        Me.m_TransitionGroupId = transitionGroupId
        Me.m_DisplayName = transitionGroupName
        Me.m_IsAuto = isAuto

    End Sub

    Public ReadOnly Property TransitionGroupId As Integer
        Get
            Return Me.m_TransitionGroupId
        End Get
    End Property

    Public ReadOnly Property DisplayName() As String
        Get
            Return m_DisplayName
        End Get
    End Property

    Friend ReadOnly Property TransitionTypes As TransitionTypeCollection
        Get
            Return Me.m_TransitionTypes
        End Get
    End Property

    Friend ReadOnly Property PrimaryTransitionTypes As TransitionTypeCollection
        Get
            Return Me.m_PrimaryTransitionTypes
        End Get
    End Property

    Friend ReadOnly Property TransitionSpreadCells As Dictionary(Of Integer, Cell)
        Get
            Return Me.m_TransitionSpreadCells
        End Get
    End Property

    Friend ReadOnly Property TransitionSpreadDistributionMap As TransitionSpreadDistributionMap
        Get
            Return Me.m_TransitionSpreadDistributionMap
        End Get
    End Property

    Friend Property PatchPrioritization As PatchPrioritization
        Get
            Return Me.m_PatchPrioritization
        End Get
        Set(value As PatchPrioritization)
            Me.m_PatchPrioritization = value
        End Set
    End Property

    Public Property HasSizeDistribution As Boolean
        Get
            Return Me.m_HasSizeDistribution
        End Get
        Set(value As Boolean)
            Me.m_HasSizeDistribution = value
        End Set
    End Property

    Public Property Order As Double
        Get
            Return Me.m_Order
        End Get
        Set(value As Double)
            Me.m_Order = value
        End Set
    End Property

    Public ReadOnly Property IsAuto As Boolean
        Get
            Return Me.m_IsAuto
        End Get
    End Property

End Class

File: Runtime/TransitionGroup.cs
// A SyncroSim Package for developing state-and-transition simulation models using ST-Sim.
// Copyright © 2007-2018 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

using System.Collections.Generic;

namespace SyncroSim.STSim
{
    public class TransitionGroup
    {
        private int m_TransitionGroupId;
        private string m_DisplayName;
        private TransitionTypeCollection m_TransitionTypes = new TransitionTypeCollection();
        private TransitionTypeCollection m_PrimaryTransitionTypes = new TransitionTypeCollection();
        private Dictionary<int, Cell> m_TransitionSpreadCells = new Dictionary<int, Cell>();
        private TransitionSpreadDistributionMap m_TransitionSpreadDistributionMap = new TransitionSpreadDistributionMap();
        private bool m_HasSizeDistribution;
        private PatchPrioritization m_PatchPrioritization;
        private double m_Order = Constants.DEFAULT_TRANSITION_ORDER;
        private bool m_IsAuto;

        public TransitionGroup(int transitionGroupId, string transitionGroupName, bool isAuto)
        {
            this.m_TransitionGroupId = transitionGroupId;
            this.m_DisplayName = transitionGroupName;
            this.m_IsAuto = isAuto;
        }

        public int TransitionGroupId
        {
            get
            {
                return this.m_TransitionGroupId;
            }
        }

        public string DisplayName
        {
            get
            {
                return m_DisplayName;
            }
        }

        internal TransitionTypeCollection TransitionTypes
        {
            get
            {
                return this.m_TransitionTypes;
            }
        }

        internal TransitionTypeCollection PrimaryTransitionTypes
        {
            get
            {
                return this.m_PrimaryTransitionTypes;
            }
        }

        internal Dictionary<int, Cell> TransitionSpreadCells
        {
            get
            {
                return this.m_TransitionSpreadCells;
            }
        }

        internal TransitionSpreadDistributionMap TransitionSpreadDistributionMap
        {
            get
            {
                return this.m_TransitionSpreadDistributionMap;
            }
        }

        internal PatchPrioritization PatchPrioritization
        {
            get
            {
                return this.m_PatchPrioritization;
            }
            set
            {
                this.m_PatchPrioritization = value;
            }
        }

        public bool HasSizeDistribution
        {
            get
            {
                return this.m_HasSizeDistribution;
            }
            set
            {
                this.m_HasSizeDistribution = value;
            }
        }

        public double Order
        {
            get
            {
                return this.m_Order;
            }
            set
            {
                this.m_Order = value;
            }
        }

        public bool IsAuto
        {
            get
            {
                return this.m_IsAuto;
            }
        }
    }
}
